Bisq Network Monitor: Babysteps

freimair opened the pull request:
**Babysteps** release of the Bisq Network Monitor proposed [here](https://github.com/bisq-network/proposals/issues/62)

freimair committed:
Basic framework for Monitor

freimair committed:
Basic configuration infrastructure available

freimair committed:
Periodical schedule

freimair committed:
Use configurable properties file

freimair committed:
Reloading config during runtime available

freimair committed:
Metrics can be disabled via config

freimair committed:
Basic configuration tests

freimair committed:
Metrics can be reenabled via config

freimair committed:
Metrics can be renamed

freimair committed:
Use lombok/slf4j for logging

freimair committed:
Configure gradle properly

freimair committed:
Introduce general purpose Tor instance

freimair committed:
Synchronize all the things

freimair committed:
Introduce reporting stub

freimair committed:
Add initial TorStartupTime metric

freimair committed:
Enhance config file documentation

freimair committed:
Cleanup

freimair committed:
Configurable socksPort for TorStartupTime Metric

freimair committed:
TorHiddenServiceStartupTime Metric available

freimair committed:
TorRoundTripTime Metric available

freimair committed:
Refactored Metrics scheduler

freimair committed:
TorRoundtripTime Metric does samples

freimair committed:
Graceful shutdown, Netlayer 0.6.2

freimair committed:
Refactored TorStartupTime

freimair committed:
Introduce reporter concept

freimair committed:
Refactored Metric

freimair committed:
Refactored metric infrastructure

freimair committed:
Graphite reporter is available

freimair committed:
Pretty up config file

cbeams commented on gradle/witness/gradle-witness.gradle:
```diff
@@ -10,72 +10,3 @@
-        'org.jetbrains.kotlin:kotlin-stdlib-common:4b161ef619eee0d1a49b1c4f0c4a8e46f4e342573efd8e0106a765f47475fe39',
-        'com.squareup.okio:okio:114bdc1f47338a68bcbc95abf2f5cdc72beeec91812f2fcd7b521c1937876266',
-    ]
-}
```
Was this removal intentional?

freimair replied:
ahm yes and no. As the monitor node is under heavy development, I removed the witness stuff as I had to fix Netlayer, tor-binary and jtorctl dependencies over and over again. I, of course, agree to supply a release with the proper checksums.

done.

cbeams commented on monitor/src/main/java/bisq/monitor/Configurable.java:
```diff
@@ -0,0 +1,58 @@
+ * Configurable is configurable.
+ * 
+ * @author Florian Reimair
+ */
```
I'd drop any boilerplate Javadoc like this and those that follow. Fine to have it if it really explains something, but otherwise it's line noise.

freimair replied:
agreed. Updated. I fear that I have been too much in a hurry to get the thing up and running :)

cbeams commented on monitor/src/main/java/bisq/monitor/Configurable.java:
```diff
@@ -0,0 +1,58 @@
+ */
+public abstract class Configurable {
+
+
```
No need for double blank lines here

freimair replied:
done

cbeams commented on monitor/src/test/java/bisq/monitor/MonitorInfrastructureTests.java:
```diff
@@ -0,0 +1,128 @@
+
+package bisq.monitor;
+
+
```
No need for double blank lines here

freimair replied:
done

cbeams commented on monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java:
```diff
@@ -0,0 +1,102 @@
+        @Override
+        public void report(long value) {
+            result = value;
+
```
```suggestion
```

freimair replied:
done

cbeams commented on monitor/src/test/java/bisq/monitor/TorHiddenServiceStartupTimeTests.java:
```diff
@@ -0,0 +1,102 @@
+
+    }
+
+    private final static File torWorkingDirectory = new File(TorHiddenServiceStartupTimeTests.class.getSimpleName());
```
Please declare fields together at top of class, not interspersed throughout methods.

freimair replied:
only the private class is declared above. Old habits. Done.

cbeams commented on monitor/src/test/java/bisq/monitor/TorStartupTimeTests.java:
```diff
@@ -0,0 +1,83 @@
+
+package bisq.monitor;
+
+
```
No need for double blank lines here

freimair replied:
done

cbeams commented on build.gradle:
```diff
@@ -311,14 +311,34 @@ configure(project(':desktop')) {
-        compile 'net.gpedro.integrations.slack:slack-webhook:1.1.1'
+        compile 'org.slf4j:slf4j-api:1.7.22'
+        compile 'ch.qos.logback:logback-core:1.1.10'
+        compile 'ch.qos.logback:logback-classic:1.1.10'
```
Since these dependencies are now repeated between `:monitor` and `:common`, their version numbers can be extracted to `$sf4jVersion` and `$logbackVersion` to avoid version drift between the two subprojects.

freimair replied:
agreed. However, as the next steps for the Monitoring Node is integrating Bisq Protocol foo, I kept the dependencies separate as the :monitor project will depend on :common eventually.

However, done.

cbeams reviewed (changes requested):
This was a very quick review from my side. Mostly style nits, but my question about the removal of gradle-witness data should be addressed before approval.

ManfredKarrer commented:
Will review once back from holiday in January.

freimair committed:
Readme, Javadoc, cleanup

freimair committed:
Dependencies and Buildfile cleanup

ManfredKarrer commented:
Is bisq.monitor.metric.Dummy.java still needed (not used yet)?

freimair committed:
Fixed compiation error as in: Commit of shame

freimair committed:
Removed unused dummy Metric

ManfredKarrer committed:
Reformat code, organize imports, fix spelling

ManfredKarrer committed:
Rename TorRoundtripTime to TorRoundTripTime

ManfredKarrer committed:
Add monitor directory as parent for test directories

- Apply reorg imports and reformat to test classes

ManfredKarrer committed:
Update gitignore

- Add /monitor/TorHiddenServiceStartupTimeTests/* and
/monitor/monitor-tor/*

ManfredKarrer committed:
Apply code inspection

- Fix spelling
- Remove exception throw clause which was never thrown
- Add not null preconditions
- Remove unneeded types in diamonds
- Add final
- Fix java doc
- Make Reporter constructor protected (abstract)

ManfredKarrer committed:
Add guava

ManfredKarrer committed:
Use 1.1.10 instead of $logbackVersion

$logbackVersion causes build errors.

ManfredKarrer committed:
Update hashes

ManfredKarrer committed:
Ignore tests

- Revert spelling error as misspelling was used for test
- Add TODO for failing test (timing issue, seems to fail just sometimes)
- Set @Disable to test classes which take longer as Tor is started

freimair committed:
Merge branch 'master' into monitor

freimair committed:
Merge branch 'monitor-nits' of git://github.com/ManfredKarrer/bisq into ManfredKarrer-monitor-nits

freimair committed:
Merge branch 'ManfredKarrer-monitor-nits' into monitor

ManfredKarrer reviewed (approved):
ACK
